Sources/Tuplet.swift

```diff
@@ -303,24 +303,15 @@ public struct Tuplet: NoteCollection {
     }
 
     private func validate() -> Bool {
-        var isValid = true
         var notesTicks = 0
         let fullTupletTicks = groupingOrder * noteDuration.ticks
         for noteCollection in notes {
-            if let tuplet = noteCollection as? Tuplet {
-                if !isValid {
-                    break
-                }
-                isValid = tuplet.validate()
+            if let tuplet = noteCollection as? Tuplet, !tuplet.validate() {
+                return false
             }
             notesTicks += noteCollection.ticks
```

Comment:
It looks like this is not exactly equivalent unless I'm missing something. If this wasn't just you trying to clean up the code and rather, fixing something, let me know.

The original code looks it will still call `notesTicks += noteCollection.ticks` on the first invalid call whereas yours returns false right away.

Reply:
I've tried to don't change the logic here, just clean up a little bit.

I don't see a huge difference here until `notesTicks` are useless if validation failed. In previous version of the code we call break if validation failed before and then returning `false` if `!isValid`. Now I just return false, because nothing can't change the final result if validation didn't pass. Ticks matter only if `isValid` is true.

Reply:
Ah. You're right. That code was weird before. lol. Thanks!